### PR TITLE
Add API endpoint for comparing Dataset Versions

### LINKED ISSUE #10888: Feature Request: Add API endpoint for comparing Dataset Versions
**Overview of the Feature Request**
Need an API endpoint that will compare two dataset versions and return a list of differences between the versions.  This is needed to support the SPA Dataset Page

**What kind of user is the feature intended for?**
(Example users roles: API User, Curator, Depositor, Guest, Superuser, Sysadmin)
API User

**What inspired the request?**
https://github.com/IQSS/dataverse-client-javascript/issues/197

**What existing behavior do you want changed?**
None

**Any brand new behavior do you want to add to Dataverse?**
New Dataverse API endpoint

**Any open or closed issues related to this feature request?**

**Are you thinking about creating a pull request for this feature?**  
Help is always welcome, is this feature something you or your organization plan to implement?

### PR DESCRIPTION
**Pull request by stevenwinship**
**What this PR does / why we need it**: Need an API endpoint that will compare two dataset versions and return a list of differences between the versions. This is needed to support the SPA Dataset Page

**Which issue(s) this PR closes**: https://github.com/IQSS/dataverse/issues/10888

- Closes #10888

**Special notes for your reviewer**:

**Suggestions on how to test this**: See IT and Unit tests

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: New API

**Is there a release notes update needed for this change?**: Included

**Additional documentation**:
https://dataverse-guide--10945.org.readthedocs.build/en/10945/

### CODE REVIEW

**Comment by stevenwinship** [comment]
> HI @stevenwinship I tried the latest version of the API, it looks good, just needs a few small changes:
> 
> * oldVersion: use 'lastUpdatedDate' rather than 'createdDate'
> * files: (added, removed or replaced):  include tags (categories) and filePath (directoryLabel)

Fixed

**Comment by pdurbin** [comment]
There's potential for merge conflicts with this PR:

- #10969

Right? 🤔

**Comment by luddaniel** [comment]
> There's potential for merge conflicts with this PR:
> 
> * [Ordering subfields while displaying dataset version differences #10969](https://github.com/IQSS/dataverse/pull/10969)
> 
> Right? 🤔

I think it's conflict-free. Maybe in IT tests. I will help if needed.
Good job @stevenwinship by the way.

File: doc/sphinx-guides/source/api/native-api.rst [hunk 1]
```diff
@@ -1642,6 +1642,25 @@ The fully expanded example above (without environment variables) looks like this
...
+  export VERSION1=:draft
+
+  curl "$SERVER_URL/api/datasets/$ID/versions/$VERSION0/compare/$VERSION1"
+
+The fully expanded example above (without environment variables) looks like this:
+
+.. code-block:: bash
+
+  curl "https://demo.dataverse.org/api/datasets/24/versions/:latest-published/compare/:draft"
+
```
Comment (sekmiller) [inline doc/sphinx-guides/source/api/native-api.rst:1663]: You should say that you need to provide an api token with view unpublished privileges to compare draft version

File: src/main/java/edu/harvard/iq/dataverse/api/Datasets.java [hunk 2]
```diff
@@ -2992,6 +2992,23 @@ public Response cleanStorage(@Context ContainerRequestContext crc, @PathParam("i
...
+    @Path("{id}/versions/{versionId1}/compare/{versionId2}")
+    public Response getCompareVersions(@Context ContainerRequestContext crc, @PathParam("id") String id,
+                                      @PathParam("versionId1") String versionId1,
+                                      @PathParam("versionId2") String versionId2,
+                                      @Context UriInfo uriInfo, @Context HttpHeaders headers) {
+        try {
+            DataverseRequest req = createDataverseRequest(getRequestUser(crc));
+            DatasetVersion dsv1 = getDatasetVersionOrDie(req, versionId1, findDatasetOrDie(id), uriInfo, headers);
+            DatasetVersion dsv2 = getDatasetVersionOrDie(req, versionId2, findDatasetOrDie(id), uriInfo, headers);
+            return ok(DatasetVersion.compareVersions(dsv1, dsv2));
```
Comment (sekmiller) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java:3006]: Should there be a test to make sure that the versions are considered in the correct order?

**Review by sekmiller — CHANGES_REQUESTED** [review]
This looks very good. Just a couple of minor nits and a reminder to bring the branch up to date with Dev. Thanks

**Review by sekmiller — APPROVED** [review]
Thanks for the updates. approved

**Pull request merged**